### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by KiraAndrianova**
Spent a half of day trying to add the code to src folder, but the code didn't work - WebDriver wasn't found while dependency was in place. So I've just create different folders for pages and tests.

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/pages/differentelementspage/DifferentElementsPage.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+
+    public void tickRadios(String word) {
+        for (WebElement w : radios) {
+            if (w.getText().contains(word)) {
+                w.click();
+            }
+        }
+    }
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw3/pages/differentelementspage/DifferentElementsPage.java]: For radioButtons it's better to find button and then click on it. In other way this step may be passed even if radioButton has not been clicked.

File: src/test/java/com/epam/tc/hw3/pages/homepage/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @FindBy(id = "frame-button")
+    private WebElement frameButton;
+
+    public List<String> getHeader() {
+        return headerMenuActual.stream().map(WebElement::getText).collect(Collectors.toList());
+    }
+
+    public List<Boolean> getListImages() {
+        return webElementListImages.stream().map(WebElement::isDisplayed).collect(Collectors.toList());
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw3/pages/homepage/HomePage.java]: Method name says that waits list of images and not list of booleans. This method should be modified. It doesn't make sense to return list of boolean.

File: src/test/java/com/epam/tc/hw3/test/DifferentElementsTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                .initElements(webDriver, LoginPage.class);
+
+        DifferentElementsPage differentElementsPage = PageFactory
+                .initElements(webDriver, DifferentElementsPage.class);
+
+        loginPage.login("Roman", "Jdi1234");
+
+        differentElementsPage.openDifferentElementsPage();
+
+        WebDriverWait wait = new WebDriverWait(webDriver, 10);
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw3/test/DifferentElementsTest.java]: This should be moved to before

File: src/test/java/com/epam/tc/hw3/test/DifferentElementsTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        wait.until(ExpectedConditions.elementToBeClickable(differentElementsPage.getMainContent()));
+
+        for (String winHandle : webDriver.getWindowHandles()) {
+            webDriver.switchTo().window(winHandle);
+        }
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw3/test/DifferentElementsTest.java]: Need to get rid of cycles in tests.

File: src/test/java/com/epam/tc/hw3/test/HomePageTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        HomePage homePage = PageFactory
+                .initElements(webDriver, HomePage.class);
+
+        List<String> expectedHeaderMenu = asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+
+        softAssertions.assertThat(homePage.getHeader())
+                .containsExactlyElementsOf(expectedHeaderMenu);
+
+        softAssertions.assertThat(homePage.getListImages())
+                .allMatch(t -> t == true)
```
Comment (Delnote) [inline src/test/java/com/epam/tc/hw3/test/HomePageTest.java]: It should be modified after getListImages changes

**Review by Delnote — CHANGES_REQUESTED** [review]
All page objects and other main stuff should be moved from 'test' section to main java section.

**Comment by KiraAndrianova** [comment]
Please check the updates

**Review by Delnote — APPROVED** [review]

**Pull request merged**